### DISCUSSION TITLE
Noob question: Why npm?

### DISCUSSION THREAD

**Discussion by harryfear** [tabler/tabler#1110] [Q&A] [answered]
Discussion closed

**Comment by schbz** [comment, marked as answer]
It's a reasonable question. I suppose it's just easier for there to be one package manager for all things JavaScript/TS, rather than adding an alternative for client side code. Since all the code running on the client side has to start out as data on the server side at some point, I don't see how there would be any advantages to having a specifically client side package manager. Can you? Also, there are plenty of really good reasons not to use public CDNs, especially for production level websites.

**Comment by jonbk** [comment]
Using NPM for installing Tabler allow me to build a very specific JS bundle with webpack. I only choose what I need from the library and webpack includes only this code to the final build. Using the CDN is downloading the whole library to the client side.

NPM is useless on the final client side for sure, but very useful for developers in the development or building processes.
